pkg/kubelet/kubelet.go

```diff
@@ -1573,26 +1573,130 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 	}
 }
 
-// Updated the Kubelet's internal pods with those provided by the update.
+func isStaticPod(pod *api.BoundPod) bool {
+	if getPodSource(pod) != ApiserverSource {
+		return true
+	}
+	return false
+}
+
+func isMirrorPod(pod *api.BoundPod) bool {
+	if value, ok := pod.Annotations[ConfigMirrorAnnotationKey]; !ok {
+		return false
+	} else {
+		return value == MirrorType
+	}
+}
+
+func (kl *Kubelet) createMirrorPod(pod *api.BoundPod) error {
```

Comment:
We should probably batch all the creation of mirror pods and do them async on the podWorkers. Doing these here will slow down pod startup latency overall. Maybe we can mark the pods as mirror pods and let syncPod() handle the create? Or just remove all static pods from here and only act on them when they're served by the API server

It can be argued that they don't happen often and shouldn't affect operation though.